### PR TITLE
Add pre commit

### PR DESCRIPTION
**Pull request by mkinney**
Partly resolves: #216 

Experimenting with pre-commit python plugin.
Just used default config.

I *think* it does not use the same flake8 config as the project uses, which is a bummer. But, it make prevent multiple commits (like 'flake8 fixes').

Thoughts?

### CODE REVIEW

**Comment by mkinney** [comment]
Grrr... if in a subdir and try to commit, the whitelist.txt from the main directory will not be used. You can experience this by going into examples and running "flake8".

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,8 +14,17 @@ git clone git@github.com:Fizzadar/pyinfra.git
 
 # Install the package in editable mode with development requirements
 pip install -e .[dev]
+
+# Configure git to use pre-commit hook
+flake8 --install-hook git
```
Comment (Fizzadar) [inline CONTRIBUTING.md]: I think this needs to be `pre-commit install`.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
```
Comment (Fizzadar) [inline .pre-commit-config.yaml]: Also needs flake8 config here I think as per https://github.com/pre-commit/pre-commit/blob/master/.pre-commit-config.yaml#L14.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mkinney) [reply CONTRIBUTING.md]: Those two commands produce different hooks. I'm ok with either one. The flake8 one was much simpler.

Comment (Fizzadar) [reply CONTRIBUTING.md]: Sounds godo to me - we'll need to delete the `.pre-commit-config.yaml` file and `pre-commit` dependency in that case as it won't be required anymore.

Comment (mkinney) [reply CONTRIBUTING.md]: Do you have a preference? I'm just experimenting.

Comment (Fizzadar) [reply CONTRIBUTING.md]: I think `flake8` for now.

File: .pre-commit-config.yaml (hunk 2 above)
Comment (mkinney) [reply .pre-commit-config.yaml]: not applicable since we're just going to try with flake8 for now...

**Pull request merged**